src/components/common/Pagination.js

```diff
@@ -26,20 +30,34 @@ class Pagination extends Component {
     this.decrementPage = this.decrementPage.bind(this);
   }
 
+  componentDidUpdate() {
+    const { pageNumber } = this.props;
+    const { currentPage } = this.state;
+    if (pageNumber !== currentPage) {
+      this.setActive({ target: { value: pageNumber } });
+      return true;
+    }
+    return false;
```

Comment:
Why is your componentDidUpdate not taking in any argument, at least a `prevProps`?
Do you think it is reasonably sufficient to make comparisons between the latest pageNumber props and the present currentPage state. Are we not risking making infinite database calls and falling into a maximum depth problem. I believe it is safe: to make a comparison between the prevProps and the latest props like so: 
`if (prevProps.pageNumber !== this.props.pageNumber)`.
Refer to: [https://reactjs.org/docs/react-component.html#componentdidupdate](url) for more on this.

May I also know why we are returning `true` or `false` on this lifeCycle method?

Reply:
<img width="955" alt="screen shot 2018-12-28 at 3 35 40 am" src="https://user-images.githubusercontent.com/30579559/50500660-2755ab80-0a53-11e9-8ae6-175737143357.png">

- Referring to the above snapshot: You seem to have queried the database twice instead of once. May I know why this is happening?
- Bug reproduction steps:
- navigate to the advance search-page
- clear the console
- `Click/toggle` the `articles/page` to `50`:
- A single click seems to have dispatched the same set of actions twice, and this happens concurrently. 

Please get this fixed.
Research more on the async/slow nature of `this.setState`. The following links might be of help.
[https://github.com/facebook/react/issues/11527#issuecomment-360199710](https://github.com/facebook/react/issues/11527#issuecomment-360199710)
[https://reactjs.org/docs/react-component.html#setstate](https://reactjs.org/docs/react-component.html#setstate)
[https://reactjs.org/docs/react-component.html#componentdidupdate](https://reactjs.org/docs/react-component.html#componentdidupdate)

Reply:
@augustineezinwa Please move your second suggestion to another thread as it has nothing to do with the original thread. I will be resolving this issue now. Thanks.